miniscope_io/stream_daq.py

```diff
@@ -370,6 +395,8 @@ def _buffer_to_frame(
             Input buffer queue.
         frame_buffer_queue : multiprocessing.Queue[ndarray]
             Output frame queue.
+        continuous : bool, optional
+            If True, continue capturing until a KeyboardInterrupt is received, by default False.
```

Comment:
It's a bit different, so I'll update this; this doesn't affect queue skipping but changes behavior when the input queue is empty. This flag keeps running even when the data stream is stopped, whereas the previous version puts sentinel in the output queue and terminates the recording.

So the previous code was:
```python
try:
    for serial_buffer in exact_iter(serial_buffer_queue.get, None):
    [processing stuff]
finally:
    [terminating stuff]
```
But this terminates if the input queue is empty, so I put it in a while loop. And this continuous is a flag to break the while loop if it's not in continuous mode.
```python
try:
    while not self.terminate.is_set():
        for serial_buffer in exact_iter(serial_buffer_queue.get, None):
    [processing stuff]
    if continuous is False:
        break
finally:
    [terminating stuff]
```

Reply:
Done

```
continuous: bool, optional
    This flag changes the termination behavior when the input queue is empty.
    In both cases the capture terminates when KeyboardInterrupt is received.
    If True, capture continues waiting when the input queue is empty.
    If false, the capture will terminate when the input queue is empty.
```

Reply:
Aha, ok yes it looked like this PR did both, added a timeout to the `put` calls (so it would skip putting the data in if the upstream queues were backed up) as well as the change to the iteration logic. Ill take another look and appreciate the updated explanation  :)